### PR TITLE
PyTorch Static-Quant and Smooth-Quant 3x API Implementation

### PR DESCRIPTION
**Pull request by violetch24**
## Type of Change
Migration static-quant and smooth-quant implementation to 3x API.

## Description

- [x] Refine weight-only configuration
  - [x] Static
  - [x] Smoothquant
- [ ] Separate public utilities from sq algos
- [ ] Unify interface for quantization
  - [ ] Static
  - [ ] Smoothquant
- [ ] Migrate algos
  - [ ] Static
  - [ ] Smoothquant
- [ ] Add UTs for each algo
  - [ ] Static
  - [ ] Smoothquant
- [x] Add UTs for configs
  - [x] Static
  - [x] Smoothquant

## Expected Behavior & Potential Risk
UT pass

### CODE REVIEW

File: neural_compressor/torch/quantization/config.py [hunk 1]
```diff
@@ -311,6 +319,205 @@ def get_default_gptq_config() -> GPTQConfig:
...
+    """Config class for static quantization."""
+
+    name = STATIC_QUANT
+    supported_configs: List[OperatorConfig] = []
+    params_list = [
+        "w_dtype",
+        "w_sym",
+        "w_granularity",
+        "w_algo",
+        "act_dtype" "act_sym",
```
Comment (xin3he) [inline neural_compressor/torch/quantization/config.py:334]: miss `,`.

File: neural_compressor/torch/quantization/config.py [hunk 2]
```diff
@@ -311,6 +319,205 @@ def get_default_gptq_config() -> GPTQConfig:
...
+    """Config class for smooth quantization."""
+
+    name = SMOOTH_QUANT
+    supported_configs: List[OperatorConfig] = []
+    params_list = [
+        "w_dtype",
+        "w_sym",
+        "w_granularity",
+        "w_algo",
+        "act_dtype" "act_sym",
```
Comment (xin3he) [inline neural_compressor/torch/quantization/config.py:418]: same bug

File: neural_compressor/torch/quantization/config.py [hunk 3]
```diff
@@ -311,6 +319,205 @@ def get_default_gptq_config() -> GPTQConfig:
...
+        }
+        self._post_init()
+
+    def to_dict(self):
+        return super().to_dict(params_list=self.params_list, operator2str=operator2str)
+
+    @classmethod
+    def from_dict(cls, config_dict):
+        return super(SmoothQuantConfig, cls).from_dict(config_dict=config_dict, str2operator=str2operator)
+
```
Comment (xin3he) [inline neural_compressor/torch/quantization/config.py:486]: remember removing these functions.

File: neural_compressor/torch/quantization/config.py [hunk 4]
```diff
@@ -311,6 +319,205 @@ def get_default_gptq_config() -> GPTQConfig:
...
+    @classmethod
+    def register_supported_configs(cls) -> List[OperatorConfig]:
+        supported_configs = []
+        # TODO(Yi)
+        linear_sq_config = SmoothQuantConfig()
+        operators = [torch.nn.Linear, torch.nn.functional.linear]
```
Comment (xin3he) [inline neural_compressor/torch/quantization/config.py:492]: please remove functional.linear. It's not workable

**Review by xin3he — APPROVED** [review]

**Comment by violetch24** [comment]
resolved in https://github.com/intel/neural-compressor/pull/1568

**Pull request closed without merging**